### DISCUSSION TITLE
Getting value of property declaration?

### DISCUSSION THREAD

**Discussion by JordanLongstaff** [LemonAppDev/konsist#619] [Ask a Question] [answered]
### Wanted end result.

I routinely declare so many class/object properties and inline their initial values, e.g.:

```
const val CONST: String = "const"
```

But for whatever reason I can't seem to find any way to fetch that initial value declaration?

### What you tried so far.

Here's a workaround I just found:

```
val propDecl: KoPropertyDeclaration = // ...
propDecl.text.endsWith("\"$expectedValue\"")
```

**Comment by nataliapeterwas** [comment, marked as answer]
Thanks for reporting this issue. For now, this is a way to go. We have a ticket in the backlog to add this, but this is a more complex problem. 

The base example, like `val sampleProperty: String = " ... " ` is easy to add, but we are considering also more complex examples, e.g.:
- delegations : 
```kotlin
val sampleProperty: String by lazy { " ... " }
```
- getter and setters : 
```kotlin
var sampleProperty: String = " ... "
    get() {
        return field
    }
    set(value) {
        field = value
    }
```

**Comment by igorwojda** [comment]
A new version `v0.13.0` has been released. It contains the ability to get a property value. 

```kotlin
Konsist
    .scopeFromProject()
    .properties()
    .assertTrue { it.value.startsWith("prefix") }
```

See https://github.com/LemonAppDev/konsist/releases/tag/v0.13.0
